src/Features/CSharp/Portable/EditAndContinue/CSharpEditAndContinueAnalyzer.cs

```diff
@@ -1333,16 +1304,49 @@ internal static TextSpan GetDiagnosticSpanImpl(SyntaxKind kind, SyntaxNode node,
                 case SyntaxKind.OrderByClause:
                     return ((OrderByClauseSyntax)node).OrderByKeyword.Span;
 
-                case SyntaxKind.AscendingOrdering:
-                case SyntaxKind.DescendingOrdering:
-                    return node.Span;
-
                 case SyntaxKind.SelectClause:
                     return ((SelectClauseSyntax)node).SelectKeyword.Span;
 
                 case SyntaxKind.GroupClause:
                     return ((GroupClauseSyntax)node).GroupKeyword.Span;
 
+                case SyntaxKind.ForEachComponentStatement:
```

Comment:
I could not discern any order and found it confusing to have them spread out. I'll revert that refactoring.